nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/NodeStatusHistoryDumpFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.controller.status.history;
+
+import java.util.Calendar;
+import java.util.Date;
+
+public class NodeStatusHistoryDumpFactory implements StatusHistoryDumpFactory {
+
+    private StatusHistoryRepository statusHistoryRepository;
+
+    @Override
+    public StatusHistoryDump create(int days) {
+        final Date now = new Date();
```

Comment:
I am not quite sure if the date calculation is the best from user perspective. Would not it make more sense to use calendar day? From user perspective it looks to be simpler to ask for the data for this day and the day before for example (numberOfDays is 2)